test/integration/scheduler_perf/util.go

```diff
@@ -166,55 +170,58 @@ func (*metricsCollector) run(ctx context.Context) {
 func (pc *metricsCollector) collect() []DataItem {
 	var dataItems []DataItem
 	for _, metric := range pc.Metrics {
-		dataItem := collectHistogram(metric, pc.labels)
-		if dataItem != nil {
-			dataItems = append(dataItems, *dataItem)
-		}
+		itemCollection := collectHistogram(metric, pc.labels)
+		dataItems = append(dataItems, itemCollection...)
 	}
 	return dataItems
 }
 
-func collectHistogram(metric string, labels map[string]string) *DataItem {
-	hist, err := testutil.GetHistogramFromGatherer(legacyregistry.DefaultGatherer, metric)
-	if err != nil {
-		klog.Error(err)
-		return nil
-	}
-	if hist.Histogram == nil {
-		klog.Errorf("metric %q is not a Histogram metric", metric)
-		return nil
+func collectHistogram(metric string, labels map[string]string) []DataItem {
+	var dataItems []DataItem
+	metricLabelMap := map[string]string{
+		FrameworkExtensionPointDurationMetricName: ExtensionPointLabelName,
 	}
-	if err := hist.Validate(); err != nil {
+	histMap, err := testutil.GetHistogramFromGatherer(legacyregistry.DefaultGatherer, metric, metricLabelMap)
+	if err != nil {
 		klog.Error(err)
 		return nil
 	}
+	for label, hist := range histMap {
+		if err := hist.Validate(); err != nil {
+			klog.Error(err)
+			return nil
+		}
 
-	q50 := hist.Quantile(0.50)
-	q90 := hist.Quantile(0.90)
-	q99 := hist.Quantile(0.95)
-	avg := hist.Average()
+		q50 := hist.Quantile(0.50)
+		q90 := hist.Quantile(0.90)
+		q99 := hist.Quantile(0.95)
+		avg := hist.Average()
 
-	// clear the metrics so that next test always starts with empty prometheus
-	// metrics (since the metrics are shared among all tests run inside the same binary)
-	hist.Clear()
+		// clear the metrics so that next test always starts with empty prometheus
+		// metrics (since the metrics are shared among all tests run inside the same binary)
+		hist.Clear()
 
-	msFactor := float64(time.Second) / float64(time.Millisecond)
+		msFactor := float64(time.Second) / float64(time.Millisecond)
 
-	// Copy labels and add "Metric" label for this metric.
-	labelMap := map[string]string{"Metric": metric}
-	for k, v := range labels {
-		labelMap[k] = v
-	}
-	return &DataItem{
-		Labels: labelMap,
-		Data: map[string]float64{
-			"Perc50":  q50 * msFactor,
-			"Perc90":  q90 * msFactor,
-			"Perc99":  q99 * msFactor,
-			"Average": avg * msFactor,
-		},
-		Unit: "ms",
+		// Copy labels and add "Metric" label for this metric.
+		labelMap := map[string]string{"Metric": label}
+		for k, v := range labels {
+			labelMap[k] = v
+		}
+		dataItem := &DataItem{
```

Comment:
use a value here: `dataItem := DataItem{`, and then in L221 `dataItems = append(dataItems, dataItem)`.

Reply:
Done